router/functionServiceMap.go

```diff
@@ -80,3 +82,24 @@ func (fmap *functionServiceMap) remove(f *metav1.ObjectMeta) error {
 	mk := keyFromMetadata(f)
 	return fmap.cache.Delete(*mk)
 }
+
+func (fmap *functionServiceMap) grabUpdateLock(f *metav1.ObjectMeta, wg *sync.WaitGroup) (*sync.WaitGroup, error) {
```

Comment:
Not sure where to put these function is better, for now just put these two here.

Reply:
Yes we should add a new cache `fmap.waitGroups` or something like that